dimagi/commcare-hq
# [enterprise] Refactor get_account_or_404 due to authorization ambiguity

## Summary
Looks like `get_account_or_404` has remained roughly in the same state since the beginning of the enterprise admin UIs.
This refactors it to be more explicit about when enterprise admin permissions checks are being made on a request vs when an account is simply being fetched.

## Safety Assurance


### Automated test coverage
While there is a test for the decorator, I'm not confident there are automated tests covering other changes (actually pretty sure there aren't).

### QA Plan
- test enterprise admin views and make sure they are only accessible by enterprise admins or accounting admins.

for both wire and non-wire/standard invoices:
- check that when viewing invoice pdfs for customer accounts, only enterprise admins or accounting admins can view
- make sure that non-customer account invoice pdfs can be viewed only by users with admin / billing privileges

### Safety story
when making any permissions changes to active areas of the codebase without solid tests, that definitely is a cue that QA is needed. so even though this change seems fairly "small" I would rather not risk a quick merge as the data accessed is very sensitive!

### Rollback instructions
